components/habits/HabitWeeklyView.tsx

```diff
@@ -0,0 +1,34 @@
+import { HabitWithHistory } from '../../src/types/habits';
+import CompleteButton from '../entries/CompleteButton';
+import useCompletionPercentages from './hooks';
+
+type HabitWeeklyViewProps = {
+  habitWithHistory: HabitWithHistory;
+  dates: string[];
+};
+
+const HabitWeeklyView = ({ habitWithHistory, dates }: HabitWeeklyViewProps) => {
+  const completionPercentagesByDay = useCompletionPercentages({
+    habitWithHistory,
+    dates,
+  });
+
+  return (
+    <tbody>
+      <tr>
+        <td>{habitWithHistory.name}</td>
+        {dates.map((date) => (
+          <td key={date}>
+            {completionPercentagesByDay[date] === 100 ? (
```

Comment:
I've seen this condition in a few components now - you may wish to extract this logic into a function that the `useCompletionPercentages` hook can supply or you can just import

Reply:
What do you mean by this?

Reply:
Really about finding a reusable way of capturing that logic - so there's one place that the logic exists, and many places that can call upon it (function is also ok for now, but it could be something that the habits API or hook returns)